addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,460 @@
+        }
+        if (error.getDate() != null) {
+            State dateTime = convertDateTimeType(error.getDate(), error.getTime());
+            updateState(CHANNEL_LAST_ERROR_DATE, dateTime);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        try {
+            RobonectConfig robonectConfig = getConfigAs(RobonectConfig.class);
+            RobonectEndpoint endpoint = new RobonectEndpoint(robonectConfig.getHost(), robonectConfig.getUser(),
+                    robonectConfig.getPassword());
+            httpClient = new HttpClient();
+            httpClient.start();
+            robonectClient = new RobonectClient(httpClient, endpoint);
+            refreshVersionInfo();
+            Runnable runnable = new Runnable() {
+                
+                private long offlineSince = -1;
+                private long offlineTriggerDealay = robonectConfig.getOfflineTimeout() * 60 * 1000;
+                private boolean offlineTimeoutTriggered = false;
+                
+                @Override
+                public void run() {
+                    try {
+                        refreshMowerInfo();
+                        updateStatus(ThingStatus.ONLINE);
+                        offlineSince = -1;
+                        offlineTimeoutTriggered = false;
+                    } catch (RobonectCommunicationException rce) {
+                        if(offlineSince < 0){
+                            offlineSince = System.currentTimeMillis();
+                        }
+                        if(!offlineTimeoutTriggered && System.currentTimeMillis() - offlineSince > offlineTriggerDealay){
+                            // trigger offline
+                            updateState(CHANNEL_MOWER_STATUS_OFFLINE_TRIGGER, new StringType("OFFLINE_TIMEOUT") );
+                            offlineTimeoutTriggered = true;
+                        }
+                        logger.debug("Failed to communicate with the mower. Taking it offline.", rce);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, rce.getMessage());
+                    } catch(com.google.gson.JsonSyntaxException jse){
+                        // the module sporadically sends invalid json responses. As this is usually recovered with the
+                        // next poll interval, we just log it to debug here.
+                        logger.debug("Failed to parse response.", jse);
+                    } catch (Exception e) {
+                        logger.error("Unexpected exception. Setting thing offline", e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+                    }
+                }
+            };
+            int pollInterval = robonectConfig.getPollInterval();
+            pollingJob = scheduler.scheduleAtFixedRate(runnable, 0, pollInterval, TimeUnit.SECONDS);
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)

Reply:
yep, done... Just wonder why I haven't seen this before